discord/ext/commands/help.py

```diff
@@ -336,88 +256,57 @@ class HelpCommand:
 
     MENTION_PATTERN = re.compile('|'.join(MENTION_TRANSFORMS.keys()))
 
-    if TYPE_CHECKING:
-        __original_kwargs__: Dict[str, Any]
-        __original_args__: Tuple[Any, ...]
-
     def __new__(cls, *args: Any, **kwargs: Any) -> Self:
-        # To prevent race conditions of a single instance while also allowing
-        # for settings to be passed the original arguments passed must be assigned
-        # to allow for easier copies (which will be made when the help command is actually called)
-        # see issue 2123
+        # we need to wrap the command callback to ensure context is correctly set.
+        functools.update_wrapper(cls._set_context, cls.command_callback)
+        cls._set_context.__qualname__ = 'command_callback'
```

Comment:
Not really a fan of this. I know this is attempting to lie to `utils.is_inside_class`. I have no idea what the wrapping is for.

Reply:
Wrapping is to have `self.context` set without requring subclasses to explicitly do so.

This was one of my own main gripes with the implimentation.

Reply:
I reverted back to overriding params in the constructor.